### ISSUE TITLE
Rule: Around() plugins

### ISSUE THREAD

**Issue by jissereitsma** [extdn/extdn-phpcs#2]
Analyse `around()` plugin methods to see if they still call upon the original callable (commonly named `$proceed()`).

**Comment by fooman** [comment]
I am not sure this is going to be automatable. If you use `around()` to change behaviour in the default implementation calling `$proceed()` is not possible.

**Comment by jissereitsma** [comment]
Correct. Some of these rules will not be usable as strict requirement, but more as something that needs to be reported. I definitely would like to know whether a plugin breaks the chain of other plugin methods plus original method (by not calling upon the `$proceed()`), while it might not be something that would define this code as bad. However, it could be debated that changing the behavior in the default implementation should not be done using a plugin but a preference instead?

**Comment by schmengler** [comment]
I think this would be a good example for an *extension interoperability* check. I would not try to incorporate this into this ruleset though, rather in a separate tool.

**Comment by schmengler** [comment]
> However, it could be debated that changing the behavior in the default implementation should not be done using a plugin but a preference instead?

Disagree. To change only one method, a plugin is better because multiple extensions may replace different methods without clashing.
